src/test/java/com/programmers/voucher/service/VoucherServiceTest.java

```diff
@@ -0,0 +1,91 @@
+import com.programmers.voucher.dto.request.VoucherCreationRequest;
+import com.programmers.voucher.repository.MemoryVoucherRepository;
+import com.programmers.voucher.repository.VoucherRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.NullSource;
+
+import java.util.List;
+
+class VoucherServiceTest {
+    public static final int FIXED_DISCOUNT_AMOUNT = 100;
+    public static final int PERCENT_DISCOUNT_AMOUNT = 10;
+    public static final String FIXED_AMOUNT_VOUCHER_TYPE = "FIXED";
+    public static final String PERCENT_DISCOUNT_VOUCHER_TYPE = "PERCENT";
+
+    private VoucherRepository voucherRepository;
+    private VoucherService voucherService;
+
+    @DisplayName("MemoryVoucherRepository, voucherService 초기화")
+    @BeforeEach
+    void init() {
+        MemoryVoucherRepository.clear();
+        voucherRepository = new MemoryVoucherRepository();
+        voucherService = new VoucherService(voucherRepository);
+    }
+
+    @DisplayName("다양한 바우처 생성 성공 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {
+            "FIXED, 100",
+            "PERCENT, 10"})
+    void createVoucher(String voucherType, long discountAmount) {
+        VoucherCreationRequest voucherCreationRequest = new VoucherCreationRequest(voucherType, discountAmount);
+        voucherService.createVoucher(voucherCreationRequest);
+    }
+
+    @DisplayName("잘못된 바우처 타입 입력시 예외 발생 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {
+            "wrongType, 100",
+            "weirdType, 10"})
+    void inputIncorrectVoucherType(String voucherType, long discountAmount) {
+        VoucherCreationRequest voucherCreationRequest = new VoucherCreationRequest(voucherType, discountAmount);
+        Assertions.assertThatThrownBy(() -> voucherService.createVoucher(voucherCreationRequest))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @DisplayName("바우처 타입을 입력하지 않을시 예외 발생 테스트")
+    @ParameterizedTest
+    @NullSource
+    void inputNullVoucherType(String voucherType) {
+        Assertions.assertThatThrownBy(() -> new VoucherCreationRequest(voucherType, 10))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("바우처 타입과 할인양을 입력해주세요.");
+    }
+
+    @DisplayName("바우처 할인양이 0일경우 예외 발생 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {
+            "FIXED, 0",
+            "PERCENT, 0"})
+    void input0VoucherDiscountAmount(String voucherType, long discountAmount) {
+        Assertions.assertThatThrownBy(() -> new VoucherCreationRequest(voucherType, discountAmount))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("바우처 타입과 할인양을 입력해주세요.");
+    }
```

Comment:
[super nit]
```suggestion
    @EnumSource(VoucherType.class)
    void input0VoucherDiscountAmount(VoucherType voucherType) {
        final long discountAmount = 0;
        String requestVoucherType = voucherType.name()
        Assertions.assertThatThrownBy(() -> new VoucherCreationRequest(requestVoucherType , discountAmount))
                .isInstanceOf(IllegalArgumentException.class)
                .hasMessage("바우처 타입과 할인양을 입력해주세요.");
    }
```
EnumSource도 알려드리기 위한 코드!